Formula/mesa.rb

```diff
@@ -31,18 +31,23 @@ class Mesa < Formula
 
   depends_on "linuxbrew/xorg/libdrm"
   depends_on "systemd" # provides libudev <= needed by "gbm"
-  depends_on "llvm@4" # failed with llvm@6
+  depends_on "llvm@4" => :build # failed with llvm@6
   depends_on "libelf" # radeonsi requires libelf when using llvm
   depends_on "linuxbrew/xorg/libomxil-bellagio"
-  depends_on "linuxbrew/xorg/wayland-protocols" => :recommended
+  depends_on "linuxbrew/xorg/wayland-protocols" => [:recommended, :build]
+  depends_on "linuxbrew/xorg/xproto" => :build
   depends_on "valgrind" => :recommended
   depends_on "linuxbrew/xorg/libglvnd" => :optional
   depends_on "linuxbrew/xorg/libva" => :recommended
-  depends_on "linuxbrew/xorg/libvdpau"
+  depends_on "linuxbrew/xorg/libvdpau" => :build
```

Comment:
Why no run-time linkage, is it header only? Is this dependency required?

Reply:
`libvdpau` is not header-only:
```
/home/linuxbrew/.linuxbrew/Cellar/libvdpau/1.1.1/lib/libvdpau.so
/home/linuxbrew/.linuxbrew/Cellar/libvdpau/1.1.1/lib/vdpau/libvdpau_trace.so
```
But mesa does not link against these. I _think_ `mesa` uses `libvdpau` to build libs like

```
/home/linuxbrew/.linuxbrew/Cellar/mesa/17.2.3_3/lib/vdpau/libvdpau_r300.so
/home/linuxbrew/.linuxbrew/Cellar/mesa/17.2.3_3/lib/vdpau/libvdpau_radeonsi.so
/home/linuxbrew/.linuxbrew/Cellar/mesa/17.2.3_3/lib/vdpau/libvdpau_nouveau.so
```

Reply:
> Is this dependency required?

yes, it is required - the build fails without it.